Error while using DNS request

- Template used

```yaml
id: cname-service-detector

info:
  name: 3rd party service checker
  author: bauthard
  severity: info


dns:
    - name: "{{FQDN}}"
      type: CNAME
      class: inet
      recursion: true
      retries: 5
      matchers-condition: or
      matchers:

      - type: word
        name: zendesk
        words:
          - "zendesk.com"

      - type: word
        name: github
        words:
          - "github.io"

      - type: word
        name: elastic
        words:
          - "elasticbeanstalk.com"

```

- Error 

```bash
                       __     _ 
     ____  __  _______/ /__  (_)
    / __ \/ / / / ___/ / _ \/ / 
   / / / / /_/ / /__/ /  __/ /  
  /_/ /_/\__,_/\___/_/\___/_/   v1
                projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] [cname-service-detector] Loaded template 3rd party service checker (@bauthard) [info]
[cname-service-detector:github] [dns] test.melbadry9.com
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x72fd10]

goroutine 50 [running]:
sync.(*Mutex).Lock(...)
        /root/sdk/go1.13.4/src/sync/mutex.go:74
github.com/projectdiscovery/nuclei/pkg/executor.(*DNSExecutor).writeOutputDNS(0xc0001d17d0, 0xc0001b9620, 0x1b, 0xc00020a0f0, 0x0, 0x0, 0x0)
        /root/go/src/github.com/projectdiscovery/nuclei/pkg/executor/output_dns.go:41 +0x5f0
github.com/projectdiscovery/nuclei/pkg/executor.(*DNSExecutor).ExecuteDNS(0xc0001d17d0, 0xc0001b9620, 0x1b, 0x0, 0x0)
        /root/go/src/github.com/projectdiscovery/nuclei/pkg/executor/executor_dns.go:87 +0x1c7
github.com/projectdiscovery/nuclei/internal/runner.(*Runner).processTemplateWithList.func1(0xc0000102d0, 0xc0001b9620, 0x1b, 0xc0000102d8, 0xc000070960, 0xc0001cf140, 0xc0001b9620, 0x1b)
        /root/go/src/github.com/projectdiscovery/nuclei/internal/runner/runner.go:188 +0x103
created by github.com/projectdiscovery/nuclei/internal/runner.(*Runner).processTemplateWithList
        /root/go/src/github.com/projectdiscovery/nuclei/internal/runner/runner.go:181 +0x469
```

--- a/pkg/executor/executor_dns.go
+++ b/pkg/executor/executor_dns.go
@@ -42,10 +42,11 @@ func NewDNSExecutor(options *DNSOptions) *DNSExecutor {
 	dnsClient := retryabledns.New(DefaultResolvers, options.DNSRequest.Retries)
 
 	executer := &DNSExecutor{
-		dnsClient:  dnsClient,
-		template:   options.Template,
-		dnsRequest: options.DNSRequest,
-		writer:     options.Writer,
+		dnsClient:   dnsClient,
+		template:    options.Template,
+		dnsRequest:  options.DNSRequest,
+		writer:      options.Writer,
+		outputMutex: &sync.Mutex{},
 	}
 	return executer
 }

